zebra-scan/src/scan.rs

```diff
@@ -90,6 +91,14 @@ pub async fn start(
     tokio::time::sleep(INITIAL_WAIT).await;
 
     loop {
+        // Do not scan and notify if we are below sapling activation height.
+        let tip_height = tip_height(state.clone()).await?;
+        if tip_height < sapling_activation_height {
+            info!("scanner is waiting for sapling activation. Current tip: {}, Sapling activation: {}", tip_height.0, sapling_activation_height.0);
+            tokio::time::sleep(CHECK_INTERVAL).await;
+            continue;
+        }
+
```

Comment:
I suggest we move this tip-checking logic above the scanning loop into its own loop to:

1. keep the scanning loop simple, and
2. not slow it down by requesting the tip in each iteration.

Reply:
Thank you for that, was done in https://github.com/ZcashFoundation/zebra/pull/8121/commits/d6c39846cf1e72318d66cec7952fab02035fd133